pkg/migration/kubernetes.go

```diff
@@ -0,0 +1,80 @@
+package migration
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// KubernetesSource is a source implementation to read resources from Kubernetes
+// cluster.
+type KubernetesSource struct {
+	index         int
+	items         []UnstructuredWithMetadata
+	dynamicClient dynamic.Interface
+}
+
+// NewKubernetesSource returns a KubernetesSource
+func NewKubernetesSource(dynamicClient dynamic.Interface, gvks []schema.GroupVersionKind) (*KubernetesSource, error) {
```

Comment:
Documentation on the parameters (especially the `gvks` parameter) would be good.